firmware/Core/Src/telecommands/telecommand_parser.c

```diff
@@ -153,6 +155,60 @@ uint8_t TCMD_get_suffix_tag_uint64(const char *str, const char *tag_name, uint64
     return 0;
 }
 
+/// @brief Searches for a `str` like `\@tag_name=xxxx`, and sets uint8_t array `xxxx` into `value_dest`.
+/// @param str The string (haystack) to search with for the tag.
+/// @param tag_name The tag to search for, including the '@' and '='.
+/// @param value_dest The destination for the value. `*value_dest` will be set.
+/// @return 0 if the tag was found successfully. >0 if the tag was not found, or there was an error.
+/// @note This function will return an error if the character after the string is not in: `#\@\0!`
```

Comment:
With the escaping, this is difficult to make sense of. Perhaps could be written as:

```
... is not within: {'#', '\@', 0x00}
```

Unless I'm misunderstanding what's going on here

Reply:
you're right, this note doesn't make much sense. Updated for clarity